src/backend/logger/uploader.ts

```diff
@@ -0,0 +1,135 @@
+import { app } from 'electron'
+import { readFile } from 'fs/promises'
+import path from 'path'
+import { z } from 'zod'
+
+import { TypeCheckedStoreBackend } from '../electron_store'
+import { getLogFile } from './logfile'
+import { logError, logInfo, LogPrefix } from './logger'
+import { sendFrontendMessage } from '../main_window'
+
+import type { UploadedLogData } from 'common/types'
+
+const uploadedLogFileStore = new TypeCheckedStoreBackend('uploadedLogs', {
+  cwd: 'store',
+  name: 'uploadedLogs',
+  accessPropertiesByDotNotation: false
+})
+
+async function sendRequestToApi(formData: FormData, url = 'https://0x0.st') {
+  return fetch(url, {
+    body: formData,
+    method: 'post',
+    headers: {
+      'User-Agent': `HeroicGamesLauncher/${app.getVersion()}`
+    }
+  }).catch((err) => {
+    logError([`Failed to send data to 0x0.st:`, err], LogPrefix.LogUploader)
+    return null
+  })
+}
+
+/**
+ * Uploads the log file of a game / runner / Heroic to https://0x0.st
+ * @param name See {@link UploadedLogData.name}
+ * @param appNameOrRunner Used to get the log file path. See {@link getLogFile}
+ * @returns `false` if an error occurred, otherwise the URL to the uploaded file and {@link UploadedLogData}
+ */
+async function uploadLogFile(
+  name: string,
+  appNameOrRunner: string
+): Promise<false | [string, UploadedLogData]> {
+  const fileLocation = getLogFile(appNameOrRunner)
+  const filename = path.basename(fileLocation)
+
+  const fileContents = await readFile(fileLocation)
+  const fileBlob = new Blob([fileContents])
+
+  const formData = new FormData()
+  formData.set('file', fileBlob, filename)
+  formData.set('expires', '24')
```

Comment:
24hs sounds too short for me, I can imagine some users sharing a log during a weekend and nobody sees it and expires

it should be easy to re-upload though, so maybe this is not really a problem

Reply:
Honestly it is rather uncommon for support requests to go unnoticed for 24 hours (and if that does happen, as you said, reuploading the file isn't hard to do)

We can easily change this if this does end up being a problem, I'd like to take the safer (in terms of server bandwidth used) route here first